packages/core/core/src/factories/apiFactory/types.ts

```diff
@@ -0,0 +1,51 @@
+import { IntegrationPlugin } from './../../utils/nuxt';
+export interface ApiClientExtensionLifecycle {
+  beforeCreate?: (config, headers?: Record<string, string>) => any;
```

Comment:
I wonder if we should transform `config` to the `{ config }` to unify it with `afterCreate` and have easy possibility to extend it without BC in the future. Wdyt?

Reply:
sure, this is just the beginning of the extendibility, I'll create many tasks related to this.